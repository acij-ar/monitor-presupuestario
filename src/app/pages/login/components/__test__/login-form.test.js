const mockPost = jest.fn();
jest.mock('axios', () => ({ post: mockPost }));

const React = require('react');
const { shallow } = require('enzyme');

const LoginForm = require('../login-form');

describe('Login form component', () => {
  beforeEach(() => {
    mockPost.mockClear();
  });

  it('should match snapshot for empty state', () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should login successfully after pressing enter + snapshot', async () => {
    mockPost.mockReturnValueOnce(Promise.resolve());
    const wrapper = shallow(<LoginForm />);
    const passwordInput = wrapper.find('input');
    await passwordInput.simulate('change', { target: { value: 'password123' } });
    await passwordInput.simulate('keydown', {key: 'Enter'});
    expect(mockPost).toHaveBeenCalledWith('/api/admin/login', { password: 'password123' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should login successfully after clicking button + snapshot', async () => {
    mockPost.mockReturnValueOnce(Promise.resolve());
    const wrapper = shallow(<LoginForm />);
    const passwordInput = wrapper.find('input');
    await passwordInput.simulate('change', { target: { value: 'password456' } });
    await wrapper.find('button').simulate('click');
    expect(mockPost).toHaveBeenCalledWith('/api/admin/login', { password: 'password456' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should fail login + snapshot', async () => {
    mockPost.mockReturnValueOnce(Promise.reject('Mocked login error'));
    const wrapper = shallow(<LoginForm />);
    const passwordInput = wrapper.find('input');
    await passwordInput.simulate('change', { target: { value: 'invalid-password' } });
    await wrapper.find('button').simulate('click');
    expect(mockPost).toHaveBeenCalledWith('/api/admin/login', { password: 'invalid-password' });
    expect(wrapper).toMatchSnapshot();
  });
});
