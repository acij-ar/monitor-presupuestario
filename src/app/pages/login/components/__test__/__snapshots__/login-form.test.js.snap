// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component should fail login + snapshot 1`] = `
<div
  className="monitor-content monitor-login"
>
  <div
    className="monitor-login-page-section"
  >
    <h2>
      Ingresar
    </h2>
    Contraseña
    <input
      autoFocus={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="invalid-password"
    />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Ingresar
    </button>
    <span
      className="monitor-login-feedback"
    >
      ❌
    </span>
  </div>
</div>
`;

exports[`Login form component should login successfully after clicking button + snapshot 1`] = `
<div
  className="monitor-content monitor-login"
>
  <div
    className="monitor-login-page-section"
  >
    <h2>
      Ingresar
    </h2>
    Contraseña
    <input
      autoFocus={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="password456"
    />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Ingresar
    </button>
    <span
      className="monitor-login-feedback"
    >
      ✅
    </span>
  </div>
</div>
`;

exports[`Login form component should login successfully after pressing enter + snapshot 1`] = `
<div
  className="monitor-content monitor-login"
>
  <div
    className="monitor-login-page-section"
  >
    <h2>
      Ingresar
    </h2>
    Contraseña
    <input
      autoFocus={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value="password123"
    />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Ingresar
    </button>
    <span
      className="monitor-login-feedback"
    >
      ✅
    </span>
  </div>
</div>
`;

exports[`Login form component should match snapshot for empty state 1`] = `
<div
  className="monitor-content monitor-login"
>
  <div
    className="monitor-login-page-section"
  >
    <h2>
      Ingresar
    </h2>
    Contraseña
    <input
      autoFocus={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="password"
      value=""
    />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Ingresar
    </button>
    <span
      className="monitor-login-feedback"
    />
  </div>
</div>
`;
